pkg/flow/internal/controller/config_export.go

```diff
@@ -0,0 +1,81 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/grafana/agent/pkg/flow/internal/dag"
+	"github.com/grafana/agent/pkg/river/ast"
+	"github.com/grafana/agent/pkg/river/diag"
+	"github.com/grafana/agent/pkg/river/vm"
+)
+
+type ExportConfigNode struct {
+	configNode ConfigNode
```

Comment:
It's useful from the sense of not copy pasting all the properties that are common. I also figured if we then have any specific properties for a specific config block type that we want to track it will be more obvious this way.

Reply:
Hm. I'd usually be in favor of being more explicit for clarity, even if it means copying a little extra code. I'll leave the final call up to you though, I don't think I'm that opinionated about this right now.

Reply:
I will do a first pass refactoring for not accessing the mutex from the specific config node types then let's see what we think. I think that will help clarify.

Reply:
I made some significant changes once I considered your comments and started cleaning up the code. The commit should document them and I'm thinking `ComponentNode` should use `sharedBlockNode` in place of the 6 properties it has in common. I didn't take this step yet without giving you a chance to consider but I can do that in addition if we think.